test/extensions/transport_sockets/internal_upstream/internal_upstream_integration_test.cc

```diff
@@ -182,5 +187,28 @@ TEST_F(InternalUpstreamIntegrationTest, BasicFlowWithoutTransportSocket) {
   EXPECT_THAT(waitForAccessLog(access_log_name_), ::testing::HasSubstr("-,-,-"));
 }
 
+TEST_F(InternalUpstreamIntegrationTest, InternalConnectionBufSizeTestDefault) {
```

Comment:
Are there tests that verify what happens if the buffer is small and the data that needs to be processed by it is larger?